olp-cpp-sdk-core/src/cache/InMemoryCache.cpp

```diff
@@ -17,104 +17,88 @@
  * License-Filename: LICENSE
  */
 
-#include <ctime>
-
 #include "InMemoryCache.h"
 
 namespace olp {
 namespace cache {
-InMemoryCache::TimeProvider& InMemoryCache::DefaultTimeProvider() {
-  static InMemoryCache::TimeProvider defaultTimeProvider = []() {
-    return std::time(nullptr);
-  };
-  return defaultTimeProvider;
-}
-
-InMemoryCache::ModelCacheCostFunc& InMemoryCache::EqualityCacheCost() {
-  static InMemoryCache::ModelCacheCostFunc equalityCacheCost =
-      [](const ItemTuple& tuple) { return 1; };
-  return equalityCacheCost;
-}
-
-InMemoryCache::ModelCacheCostFunc& InMemoryCache::PresetModelSizeCacheCost() {
-  static InMemoryCache::ModelCacheCostFunc presetModelSizeCacheCost =
-      [](const InMemoryCache::ItemTuple& tuple) {
-        auto result = std::get<3>(tuple);
-        if (result == 0) result = 1;
-        return result;
-      };
-  return presetModelSizeCacheCost;
+namespace {
+inline bool HasExpiry(time_t expiry_seconds) {
+  return (expiry_seconds != InMemoryCache::kExpiryMax);
 }
+}  // namespace
 
-InMemoryCache::InMemoryCache(size_t maxSize, ModelCacheCostFunc cacheCost,
-                             TimeProvider timeProvider)
-    : m_itemTuples(maxSize, std::move(cacheCost)),
-      m_timeProvider(std::move(timeProvider)) {
-  m_itemTuples.SetEvictionCallback(
+InMemoryCache::InMemoryCache(size_t max_size, ModelCacheCostFunc cache_cost,
+                             TimeProvider time_provider)
+    : item_tuples_(max_size, std::move(cache_cost)),
+      time_provider_(std::move(time_provider)) {
```

Comment:
```suggestion
      time_provider_(std::move(time_provider)) {
```
It is out of the scope of this ticket - but what if user pass a `nullptr` TimeProvider - should we roll back to the `DefaultTimeProvider` in that case?

Reply:
Considering it is a private class and not exposed trough public includes I would assume we are the users and we should avoid that. Either way we have a issue in the backlog where we need to expose this so that we can inject a TimeProvider from ourside as default one does not work on most automotive use-cases.